hs_core/models.py

```diff
@@ -2119,12 +2161,16 @@ def check_irods_files(self, stop_on_error=False, log_errors=True,
                     errors.extend(error2)
                     ecount += ecount2
 
-            # Step 1: does every file here refer to an existing file in iRODS?
+            # Step 2: does every file here refer to an existing file in iRODS?
             for f in self.files.all():
                 if not istorage.exists(f.storage_path):
                     ecount += 1
                     msg = "check_irods_files: django file {} does not exist in iRODS"\
                         .format(f.storage_path)
+                    if clean_django:
+                        delete_resource_file(self.short_id, f.short_path, self.creator,
+                                             delete_logical_file=False)
```

Comment:
@alvacouch OK. Sounds good. The only thing I am wondering about now is that after the ```bad``` physical file references are deleted as part of this cleanup, are we going to leave some dangling logical file records around in Django DB if these clean-up resources happen to belong to composite resource type? If so, for the next release, are these dangling logical file records going to be cleaned up?